osh/hub/scripts/osh-retention

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# SPDX-License-Identifier: GPL-3.0-or-later
+# SPDX-FileCopyrightText: Copyright contributors to the OpenScanHub project.
+
+"""
+Script for cron that performs the retention policy
+"""
+import datetime
+import logging
+import os
+import shutil
+from datetime import date
+
+logger = logging.getLogger(__name__)
+
+os.environ['DJANGO_SETTINGS_MODULE'] = 'osh.hub.settings'
+
+
+def deleteTasksResults(tasksToDelete):
+    from kobo.hub import models
+    for task in tasksToDelete:
+        try:
+            pathToDelete = models.Task.get_task_dir(task.id)
+            shutil.rmtree(pathToDelete)
+        except OSError as e:
+            logging.error("Error deleting tasks folder: %s", e)
+
+
+# In order to identify personal tasks, we are using the owner field. If the owner doesn't contain any / or @, we assume
+# that it is a personal task.
+def identifyPersonalTask(owner):
+    # We could use regexps here if we want to make this more complex but for this specific case
+    # built-in string methods are enough, and we are more perfomance-oriented.
+    return "/" not in owner and "@" not in owner
+
+
+def main():
+    import django
+    django.setup()
+
+    # Retention policy Task
+    logger.info("Starting Retention Policy Enforcement")
+
+    from osh.hub.scan import models
+    tasks = models.Task.objects.all()
+
+    # ---------- Section to take parameters from DB
+
+    retentionPersonalTasks = models.RetentionPolicySetting.objects.get_days_from_category("PersonalScan")
+    retentionFailedStatus = models.RetentionPolicySetting.objects.get_days_from_category("FailedStatus")
+
+    # ---------- End Section to take parameters from DB
+
+    eligiblePersonalTasks = []
+    eligibleFailedStatus = []
+
+    # We identify which scans are eligible to be deleted
+    today = date.today()
+    for t in tasks:
+        # At the moment, we only have 2 rules for retention settings:
+        # Personal Scans
+        # Failed status
+        calculatedDatePersonalTasks = (t.dt_created + datetime.timedelta(days=retentionPersonalTasks)).date()
+        calculatedDateStatusFailed = (t.dt_created + datetime.timedelta(days=retentionFailedStatus)).date()
+
+        if identifyPersonalTask(t.owner.username):
+            if calculatedDatePersonalTasks <= today:
```

Comment:
Generally two completely tested `if` statements can be combined into a single one using `and`.  Doing so makes control flow more simple and it also cuts-down code indentation:
```suggestion
        if identifyPersonalTask(t.owner.username) and calculatedDatePersonalTasks <= today:
```

The same holds also for the nested `elif` and `if` below.